<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ObjectPropertiesDialog</class>
 <widget class="QDialog" name="ObjectPropertiesDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>894</width>
    <height>496</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Object Properties</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../../../resources.qrc">
    <normaloff>:/qtgameengine/icons/object-file.png</normaloff>:/qtgameengine/icons/object-file.png</iconset>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="0,0,1">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <layout class="QFormLayout" name="formLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="labelName">
         <property name="text">
          <string>&amp;Name:</string>
         </property>
         <property name="buddy">
          <cstring>lineEditName</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="lineEditName">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>The name of the object</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBoxSprite">
       <property name="title">
        <string>Sprite</string>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QLabel" name="labelSpritePreview">
          <property name="minimumSize">
           <size>
            <width>32</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>32</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="alignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <widget class="QLineEditWithMenu" name="lineEditSprite">
              <property name="toolTip">
               <string>The sprite used to represent the object</string>
              </property>
              <property name="readOnly">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="toolButtonSprite">
              <property name="text">
               <string>...</string>
              </property>
              <property name="popupMode">
               <enum>QToolButton::InstantPopup</enum>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_8">
            <item>
             <widget class="QPushButton" name="pushButtonNewSprite">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="toolTip">
               <string>Create a new sprite for this object</string>
              </property>
              <property name="text">
               <string>New</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pushButtonEditSprite">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="toolTip">
               <string>Edit the sprite for this object</string>
              </property>
              <property name="text">
               <string>Edit</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QCheckBox" name="checkBoxVisible">
         <property name="toolTip">
          <string>Whether the object is visible</string>
         </property>
         <property name="text">
          <string>Visible</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkBoxSolid">
         <property name="toolTip">
          <string>Whether the object is solid</string>
         </property>
         <property name="text">
          <string>Solid</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QFormLayout" name="formLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="labelDepth">
         <property name="text">
          <string>Depth:</string>
         </property>
         <property name="buddy">
          <cstring>spinBoxDepth</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSpinBox" name="spinBoxDepth">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>The depth at which the object is drawn</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QCheckBox" name="checkBoxPersistent">
         <property name="toolTip">
          <string>Wheter the object is persistent between rooms</string>
         </property>
         <property name="text">
          <string>Persistent</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="labelParent">
         <property name="text">
          <string>Parent:</string>
         </property>
         <property name="buddy">
          <cstring>lineEditParent</cstring>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLineEdit" name="lineEditParent">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>The parent from which events and actions are</string>
           </property>
           <property name="text">
            <string>&lt;no parent&gt;</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="toolButtonParent">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="3" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLineEdit" name="lineEditMask">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>The image used as a collision mask for the object</string>
           </property>
           <property name="text">
            <string>&lt;same as sprite&gt;</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="toolButtonMask">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="labelMask">
         <property name="text">
          <string>Mask:</string>
         </property>
         <property name="buddy">
          <cstring>lineEditMask</cstring>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <spacer name="verticalSpacer_2">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Minimum</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>10</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="pushButtonShowInformation">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>Show the information about the object</string>
       </property>
       <property name="text">
        <string>Show Information</string>
       </property>
       <property name="icon">
        <iconset resource="../../../resources.qrc">
         <normaloff>:/qtgameengine/icons/info.png</normaloff>:/qtgameengine/icons/info.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="standardButtons">
        <set>QDialogButtonBox::Ok</set>
       </property>
       <property name="centerButtons">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QLabel" name="labelEvents">
       <property name="text">
        <string>Events:</string>
       </property>
       <property name="buddy">
        <cstring>listViewEvents</cstring>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QListView" name="listViewEvents">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Ignored" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="contextMenuPolicy">
        <enum>Qt::CustomContextMenu</enum>
       </property>
       <property name="iconSize">
        <size>
         <width>16</width>
         <height>16</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButtonAddEvent">
       <property name="toolTip">
        <string>Click here to add another event</string>
       </property>
       <property name="text">
        <string>Add Event</string>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <item>
        <widget class="QPushButton" name="pushButtonDeleteEvent">
         <property name="toolTip">
          <string>Delete the currently selected event</string>
         </property>
         <property name="text">
          <string>Delete</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButtonChangeEvent">
         <property name="toolTip">
          <string>Change which event has this behaviour</string>
         </property>
         <property name="text">
          <string>Change</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="ActionsContainerWidget" name="actionsWidget" native="true"/>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ActionsContainerWidget</class>
   <extends>QWidget</extends>
   <header>widgets/actionscontainerwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QLineEditWithMenu</class>
   <extends>QLineEdit</extends>
   <header>widgets/qlineeditwithmenu.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>ObjectPropertiesDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>138</x>
     <y>511</y>
    </hint>
    <hint type="destinationlabel">
     <x>423</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>ObjectPropertiesDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>138</x>
     <y>511</y>
    </hint>
    <hint type="destinationlabel">
     <x>423</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
